deepchem/models/jax_models/jax_model.py

```diff
@@ -0,0 +1,373 @@
+import numpy as np
+import time
+import logging
+try:
+  from collections.abc import Sequence as SequenceCollection
+except:
+  from collections import Sequence as SequenceCollection
+
+from deepchem.data import Dataset
+from deepchem.models.models import Model
+from deepchem.models.losses import Loss
+from deepchem.models.optimizers import Optimizer
+from typing import Any, Callable, Iterable, List, Optional, Tuple, Union
+from deepchem.utils.typing import LossFn
+
+# JAX depend
+try:
+  import jax.numpy as jnp
+  import jax
+  import haiku as hk
+  import optax
+except:
+  raise ImportError("This class requires Jax, haiku and optax installed.")
+
+import warnings
+
+logger = logging.getLogger(__name__)
+
+
+class JaxModel(Model):
+  """This is a DeepChem model implemented by a Jax Model
+
+  Here is a simple example of that uses JaxModel to train a
+  Haiku (JAX Neural Network Library) based model on deepchem
+  dataset.
```

Comment:
Can we add a bit more documentation here summarizing the 3 points you brought up during the meeting that discuss why a user would be interested in using a `JaxModel`?

The easy distributed training in particular seems like a great advantage to me.

Reply:
Yeah sure, I will add them. But i think there is a lot more to add in this section and hoping to add it in the end.